README_cmake.md

```diff
@@ -751,6 +751,7 @@ Variables that control package loading:
 | Variable             | Description                                                                                                                                                                   |
 |----------------------|-------------------------------------------------------------------------------------------------------------------------------------------------------------------------------|
 | `Halide_SHARED_LIBS` | override `BUILD_SHARED_LIBS` when loading the Halide package via `find_package`. Has no effect when using Halide via `add_subdirectory` as a Git or `FetchContent` submodule. |
+| `Halide_NO_THREADS` | skip checking dependency of Threads libraray. Should be set if your toolchain does not support it (e.g. baremetal). In such a situation, some of the Halide targets using Threads (e.g. `Halide::RunGenMain`) cannot be built. |
```

Comment:
This comment makes me think that the correct solution is to put something more like

```
set(Halide_NO_THREADS "@HALIDE_NO_THREADS@")
```

In the package config file and to make sure early in the build that the variable has a value (to avoid definedness checks)

Reply:
I'm not sure if I understand the points correctly. Should `Halide_NO_THREADS` be set inside package config file depending on find_package(Threads) fails or not, rather than by the user calling find_package(Halide or HalideHelpers)? Is that what you mean to say?

My assumption when I uploaded this PR is the usage shown in #7286.
About the dependency, the targets which depends on Threads seem to be `Halide::Halide`, `Halide::Generator` and `Halide::RunGenMain`. And `add_halide_runtime` links Threads to application runtime.

Reply:
Let me ask my question a different way: would re-naming this variable to `Halide_RUNTIME_NO_THREADS` more accurately describe its use?

It doesn't seem valid to exclude `find_dependency(Threads)` on the same path that loads `Halide::Halide`.

Reply:
I agree with your points. The problem is `find_package(HalideHelpers)` for runtime and `find_package(Halide)` for Generator/libHalide go through the same path.
What we want to do is to generate runtime and function (image processing) even if toolchain does not have `Threads`. So we need some mechanism for `find_package` that doesn't break in that particular use case.